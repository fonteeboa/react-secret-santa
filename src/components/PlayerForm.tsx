import React from 'react';
import { Input, Button, Form } from 'antd';
import { useTranslation } from 'react-i18next';
import { PlayerFormProps } from '../domain';

const PlayerForm: React.FC<PlayerFormProps> = ({ onAddParticipant }) => {
    const [form] = Form.useForm();
    const { t } = useTranslation();

    const handleFinish = (values: { name: string }) => {
        onAddParticipant({ name: values.name });
        form.resetFields();
    };

    return (
        <Form form={form} layout="inline" onFinish={handleFinish}>
            <Form.Item
                name="name"
                rules={[{ required: true, message: t('invalid_name') }]}
            >
                <Input placeholder={t('name')} />
            </Form.Item>
            <Form.Item>
                <Button type="primary" htmlType="submit">
                    {t('add_player')}
                </Button>
            </Form.Item>
        </Form>
    );
};

export default PlayerForm;
