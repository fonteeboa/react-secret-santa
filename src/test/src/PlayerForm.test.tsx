import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import PlayerForm from '../../components/PlayerForm';
import { useTranslation } from 'react-i18next';
import '../mocks/matchMediaMock';

// Mock do hook useTranslation
jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (key: string) => {
      const translations: { [key: string]: string } = {
        name: 'Nome',
        add_player: 'Adicionar Jogador',
        invalid_name: 'Por favor, insira o nome!',
      };
      return translations[key];
    },
  }),
}));

describe('PlayerForm Component', () => {
  const onAddParticipant = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders the form correctly', () => {
    render(<PlayerForm onAddParticipant={onAddParticipant} />);

    expect(screen.getByPlaceholderText('Nome')).toBeInTheDocument();
    expect(screen.getByText('Adicionar Jogador')).toBeInTheDocument();
  });

  test('shows error message when trying to submit empty form', async () => {
    render(<PlayerForm onAddParticipant={onAddParticipant} />);

    const button = screen.getByText('Adicionar Jogador');
    fireEvent.click(button);

    expect(await screen.findByText('Por favor, insira o nome!')).toBeInTheDocument();
    expect(onAddParticipant).not.toHaveBeenCalled();
  });

  test('matches the snapshot', () => {
    const { asFragment } = render(<PlayerForm onAddParticipant={onAddParticipant} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
