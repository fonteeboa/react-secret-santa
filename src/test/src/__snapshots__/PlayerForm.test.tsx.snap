// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerForm Component matches the snapshot 1`] = `
<DocumentFragment>
  <form
    class="ant-form ant-form-inline css-dev-only-do-not-override-1uq9j6g"
  >
    <div
      class="ant-form-item css-dev-only-do-not-override-1uq9j6g"
    >
      <div
        class="ant-row ant-form-item-row css-dev-only-do-not-override-1uq9j6g"
      >
        <div
          class="ant-col ant-form-item-control css-dev-only-do-not-override-1uq9j6g"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                aria-required="true"
                class="ant-input css-dev-only-do-not-override-1uq9j6g ant-input-outlined"
                id="name"
                placeholder="Nome"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-form-item css-dev-only-do-not-override-1uq9j6g"
    >
      <div
        class="ant-row ant-form-item-row css-dev-only-do-not-override-1uq9j6g"
      >
        <div
          class="ant-col ant-form-item-control css-dev-only-do-not-override-1uq9j6g"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <button
                class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-primary"
                type="submit"
              >
                <span>
                  Adicionar Jogador
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
